Terminal.Gui/ConsoleDrivers/NetDriver.cs

```diff
@@ -300,808 +304,224 @@ void GetConsoleInputType (ConsoleKeyInfo consoleKeyInfo)
-						break;
-					case 35:
-					case 39:
-					case 43:
-					case 47:
-					case 55:
-					case 59:
-					case 63:
-						buttonState = MouseButtonState.ReportMousePosition;
-						break;
-					case 64:
-						buttonState = MouseButtonState.ButtonWheeledUp;
-						break;
-					case 65:
-						buttonState = MouseButtonState.ButtonWheeledDown;
-						break;
-					case 68:
-					case 72:
-					case 80:
-						buttonState = MouseButtonState.ButtonWheeledLeft;       // Shift/Ctrl+ButtonWheeledUp
-						break;
-					case 69:
-					case 73:
-					case 81:
-						buttonState = MouseButtonState.ButtonWheeledRight;      // Shift/Ctrl+ButtonWheeledDown
-						break;
-					}
-					// Modifiers.
-					switch (buttonCode) {
-					case 8:
-					case 9:
-					case 10:
-					case 43:
-						buttonState |= MouseButtonState.ButtonAlt;
-						break;
-					case 14:
-					case 47:
-						buttonState |= MouseButtonState.ButtonAlt | MouseButtonState.ButtonShift;
-						break;
-					case 16:
-					case 17:
-					case 18:
-					case 51:
-						buttonState |= MouseButtonState.ButtonCtrl;
-						break;
-					case 22:
-					case 55:
-						buttonState |= MouseButtonState.ButtonCtrl | MouseButtonState.ButtonShift;
-						break;
-					case 24:
-					case 25:
-					case 26:
-					case 59:
-						buttonState |= MouseButtonState.ButtonAlt | MouseButtonState.ButtonCtrl;
-						break;
-					case 30:
-					case 63:
-						buttonState |= MouseButtonState.ButtonCtrl | MouseButtonState.ButtonShift | MouseButtonState.ButtonAlt;
-						break;
-					case 32:
-					case 33:
-					case 34:
-						buttonState |= MouseButtonState.ReportMousePosition;
-						break;
-					case 36:
-					case 37:
-						buttonState |= MouseButtonState.ReportMousePosition | MouseButtonState.ButtonShift;
-						break;
-					case 39:
-					case 68:
-					case 69:
-						buttonState |= MouseButtonState.ButtonShift;
-						break;
-					case 40:
-					case 41:
-					case 42:
-						buttonState |= MouseButtonState.ReportMousePosition | MouseButtonState.ButtonAlt;
-						break;
-					case 45:
-					case 46:
-						buttonState |= MouseButtonState.ReportMousePosition | MouseButtonState.ButtonAlt | MouseButtonState.ButtonShift;
-						break;
-					case 48:
-					case 49:
-					case 50:
-						buttonState |= MouseButtonState.ReportMousePosition | MouseButtonState.ButtonCtrl;
-						break;
-					case 53:
-					case 54:
-						buttonState |= MouseButtonState.ReportMousePosition | MouseButtonState.ButtonCtrl | MouseButtonState.ButtonShift;
-						break;
-					case 56:
-					case 57:
-					case 58:
-						buttonState |= MouseButtonState.ReportMousePosition | MouseButtonState.ButtonCtrl | MouseButtonState.ButtonAlt;
-						break;
-					case 61:
-					case 62:
-						buttonState |= MouseButtonState.ReportMousePosition | MouseButtonState.ButtonCtrl | MouseButtonState.ButtonShift | MouseButtonState.ButtonAlt;
-						break;
-					}
+			MouseButtonState mbs = default;
+			foreach (var flag in mouseFlags.GetUniqueFlags ()) {
```

Comment:
Every one of these appears to have the same name. 

Could this loop be replaced with someting that used `Enum.GetNames` to reduce the amount of code here (and thus reduce chance of errors)?

Reply:
Done.